content/master/guides/composition-functions.md

```diff
@@ -0,0 +1,1002 @@
+            engineVersion: "12"
+      patches:
+        - fromFieldPath: spec.parameters.storageGB
+          toFieldPath: spec.forProvider.allocatedStorage
+      connectionDetails:
+        - type: FromFieldPath
+          name: username
+          fromFieldPath: spec.forProvider.username
+        - type: FromConnectionSecretKey
+          name: password
+          fromConnectionSecretKey: attribute.password
+  functions:
+  - name: quotable
+    type: Container
+    container:
+      image: xpkg.upbound.io/example-org/xfn-quotable-simple:v0.1.0
+      network:
+        policy: Runner
+```
+
+### Tips for New Functions
+
+Here are some things to keep in mind when building a Composition Function:
+
+* Your Function may be running as part of a pipeline. This means it's very
+  important that your Function pass through any desired state that it is
+  unconcerned with. If your Function is passed a desired composed resource and
+  does not return that composed resource in its output, it will be deleted.
+  Crossplane considers the desired state of the XR and any composed resources to
+  be whatever `FunctionIO` is returned by the last Function in the pipeline.
+* Crossplane won't set a `metadata.name` for your desired resources resources.
+  It's a good practice to match P&T Composition's behavior by setting
+  `metadata.generateName: "name-of-the-xr-"` for any new desired resources.
+* Don't blindly add new entries to the desired resources array every time your
+  function is invoked. Remember to check whether your desired resource is
+  already in the `observed` and/or `desired` objects. You may need to update it
+  rather than create it.
+* Don't bypass providers. Composition Functions are designed to tell Crossplane
+  how to orchestrate managed resources - not to directly orchestrate external
+  systems.
+* Include your function name and version in any results you return to aid in
+  debugging.
+* Write tests for your function! Pass it a `FunctionIO` on stdin in and ensure
+  it returns the expected `FunctionIO` on stdout.
+* Keep your Functions fast and lightweight. Remember that Crossplane will run
+  them approximately once every 30-60 seconds.
+
+## The xfn Runner
+
+Composition Function runners are designed to be pluggable. Each time Crossplane
+needs to invoke a Composition Function it makes a gRPC call to a configurable
+endpoint. The default, reference Composition Function runner is named `xfn`.
+
+{{< hint "note" >}}
+The default runner endpoint is `unix-abstract:crossplane/fn/default.sock`. It's
+possible to run Functions using a different runner, for example:
+
+```yaml
+  functions:
+  - name: my-cool-Function
+    type: Container
+    container:
+      image: xkpg.io/my-cool-Function:0.1.0
+      runner:
+        endpoint: unix-abstract:/your/custom/runner.sock
+```
+
+Currently Crossplane uses unauthenticated, unencrypted gRPC requests to run
+Functions, so requests should not be sent over the network. Encryption and
```

Comment:
```suggestion
Functions. Requests shouldn't be sent over the network. Encryption and
```

Reply:
I'd prefer to keep this as once sentence but I've used a contraction.